crates/uv-configuration/src/extras.rs

```diff
@@ -6,12 +8,15 @@ pub enum ExtrasSpecification {
     None,
     All,
     Some(Vec<ExtraName>),
+    Exclude(HashSet<ExtraName>),
 }
 
 impl ExtrasSpecification {
     /// Determine the extras specification to use based on the command-line arguments.
-    pub fn from_args(all_extras: bool, extra: Vec<ExtraName>) -> Self {
-        if all_extras {
+    pub fn from_args(all_extras: bool, no_extra: Vec<ExtraName>, extra: Vec<ExtraName>) -> Self {
+        if all_extras && !no_extra.is_empty() {
+            ExtrasSpecification::Exclude(HashSet::from_iter(no_extra))
+        } else if all_extras {
             ExtrasSpecification::All
         } else if extra.is_empty() {
             ExtrasSpecification::None
```

Comment:
Maybe we filter `extra` in the `else` branch here, to remove anything in `no_extra`?